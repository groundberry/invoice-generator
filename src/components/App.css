.App-form {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 50% 50%;
  margin: 0 auto;
  max-width: 80%;
}

.App-form-info-from {
  grid-column: 1;
  grid-row: 1;
}

.App-form-info-to {
  grid-column: 2;
  grid-row: 1;
}

.App-invoice-info {
  grid-column: 1;
  grid-row: 2;
}

.App-invoice-details {
  grid-column: 2;
  grid-row: 2;
}

