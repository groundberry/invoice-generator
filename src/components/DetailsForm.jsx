import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const styles = theme => ({
  textField: {
    display: "flex"
  }
});

class DetailsForm extends React.Component {
  render() {
    const { classes, name, email, street, city, zipCode, phone } = this.props;

    return (
      <form>
        <TextField
          label="Name"
          className={classes.textField}
          name="name"
          value={name}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Email"
          className={classes.textField}
          name="email"
          value={email}
          onChange={this.handleChange}
          type="email"
          autoComplete="email"
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Street"
          className={classes.textField}
          name="street"
          value={street}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="City"
          className={classes.textField}
          name="city"
          value={city}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Zip code"
          className={classes.textField}
          name="zipCode"
          value={zipCode}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Phone"
          className={classes.textField}
          name="phone"
          value={phone}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
      </form>
    );
  }

  handleChange = evt => {
    const { name, value } = evt.target;
    this.props.onChange({
      [name]: value
    });
  };
}

DetailsForm.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string,
  email: PropTypes.string,
  street: PropTypes.string,
  city: PropTypes.string,
  zipCode: PropTypes.string,
  phone: PropTypes.string,
  onChange: PropTypes.func
};

export default withStyles(styles)(DetailsForm);
