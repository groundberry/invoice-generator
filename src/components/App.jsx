import React from "react";
import DetailsForm from "./DetailsForm";
import "./App.css";

class App extends React.Component {
  constructor() {
    super();

    this.state = {
      from: {
        name: "",
        email: "",
        street: "",
        city: "",
        zipCode: "",
        phone: ""
      },
      to: {
        name: "",
        email: "",
        street: "",
        city: "",
        zipCode: "",
        phone: ""
      }
    };
  }

  render() {
    const { from, to } = this.state;

    return (
      <div className="App">
        <div className="App-form">
          <div className="App-form-info-from">
            <h2>From</h2>
            <DetailsForm
              name={from.name}
              email={from.email}
              street={from.street}
              city={from.city}
              zipCode={from.zipCode}
              phone={from.phone}
              onChange={this.handleChangeFrom}
            />
          </div>
          <div className="App-form-info-to">
            <h2>To</h2>
            <DetailsForm
              email={to.email}
              street={to.street}
              city={to.city}
              name={to.name}
              zipCode={to.zipCode}
              phone={to.phone}
              onChange={this.handleChangeTo}
            />
          </div>
        </div>
      </div>
    );
  }

  handleChangeFrom = newAttribute => {
    this.setState(prevState => ({
      from: {
        ...prevState.from,
        ...newAttribute
      }
    }));
  };

  handleChangeTo = newAttribute => {
    this.setState(prevState => ({
      to: {
        ...prevState.to,
        ...newAttribute
      }
    }));
  };
}

export default App;
