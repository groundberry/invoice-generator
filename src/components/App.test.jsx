import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import DetailsForm from "./DetailsForm";

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when invoice from changes", () => {
    beforeEach(() => {
      const data = {
        name: "John",
        email: "john@doe.com",
        street: "1st Ave",
        city: "Seattle",
        zipCode: "09112",
        phone: "555 666 7777"
      };

      wrapper
        .find(DetailsForm)
        .at(0)
        .simulate("change", data);
    });

    it("renders with the new from", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when invoice to changes", () => {
    beforeEach(() => {
      const data = {
        name: "Jane",
        email: "jane@doe.com",
        street: "2nd Ave",
        city: "Seattle",
        zipCode: "09112",
        phone: "555 666 8888"
      };

      wrapper
        .find(DetailsForm)
        .at(1)
        .simulate("change", data);
    });

    it("renders with the new to", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
