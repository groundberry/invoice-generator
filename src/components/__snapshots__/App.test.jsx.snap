// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <div
    className="App-form"
  >
    <div
      className="App-form-info-from"
    >
      <h2>
        From
      </h2>
      <WithStyles(DetailsForm)
        city=""
        email=""
        name=""
        onChange={[Function]}
        phone=""
        street=""
        zipCode=""
      />
    </div>
    <div
      className="App-form-info-to"
    >
      <h2>
        To
      </h2>
      <WithStyles(DetailsForm)
        city=""
        email=""
        name=""
        onChange={[Function]}
        phone=""
        street=""
        zipCode=""
      />
    </div>
  </div>
</div>
`;

exports[`App when invoice from changes renders with the new from 1`] = `
<div
  className="App"
>
  <div
    className="App-form"
  >
    <div
      className="App-form-info-from"
    >
      <h2>
        From
      </h2>
      <WithStyles(DetailsForm)
        city="Seattle"
        email="john@doe.com"
        name="John"
        onChange={[Function]}
        phone="555 666 7777"
        street="1st Ave"
        zipCode="09112"
      />
    </div>
    <div
      className="App-form-info-to"
    >
      <h2>
        To
      </h2>
      <WithStyles(DetailsForm)
        city=""
        email=""
        name=""
        onChange={[Function]}
        phone=""
        street=""
        zipCode=""
      />
    </div>
  </div>
</div>
`;

exports[`App when invoice to changes renders with the new to 1`] = `
<div
  className="App"
>
  <div
    className="App-form"
  >
    <div
      className="App-form-info-from"
    >
      <h2>
        From
      </h2>
      <WithStyles(DetailsForm)
        city=""
        email=""
        name=""
        onChange={[Function]}
        phone=""
        street=""
        zipCode=""
      />
    </div>
    <div
      className="App-form-info-to"
    >
      <h2>
        To
      </h2>
      <WithStyles(DetailsForm)
        city="Seattle"
        email="jane@doe.com"
        name="Jane"
        onChange={[Function]}
        phone="555 666 8888"
        street="2nd Ave"
        zipCode="09112"
      />
    </div>
  </div>
</div>
`;
